sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubMessageToRowTest.java

```diff
@@ -131,12 +137,7 @@ public void testSendsInvalidToDLQ() {
                     message(4, map("bttr", "vbl"), "{ \"name\" : \"baz\", \"id\" : 5 }")))
             .apply(
                 "convert",
-                ParDo.of(
-                        PubsubMessageToRow.builder()
-                            .messageSchema(messageSchema)
-                            .useDlq(true)
-                            .useFlatSchema(false)
-                            .build())
```

Comment:
Is it possible to just apply the `PubsubMessageToRow` transform here like you did in the above test? I think `NestedSchemaPubsubMessageToRow` should be a private implementation detail.

Reply:
I agree that it would be better to test against PubsubMessageToRow.

I am curious whether you have any idea how PubsubIO could be mocked or not given it is a private class.

Reply:
Sorry I'm not sure I follow. Why do you need to mock PubsubIO?

Reply:
if dead letter queue is enabled, there will be request being sent to the queue.

which you could find at https://github.com/apache/beam/blob/master/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubIOJsonTable.java#L182

Reply:
Ahh sorry I see - because the write to dead letter queue is created inside `PubsubMessageToRow` now.

Could you keep the writing to dead letter queue in `PubsubIOJsonTable` if you make `PubsubMessageToRow` expand to a `PCollectionTuple`? Then all these tests would just have to make assertions about both `outputs.get(MAIN_TAG)` and `outputs.get(DLQ_TAG)`

Reply:
Thanks!